### ISSUE TITLE
Provide URL that works with NagStaMon

### ISSUE THREAD

**Issue by arthurlogilab** [shinken-monitoring/mod-webui#414]
I find that using a widget sur as NagStaMon https://nagstamon.ifw-dresden.de/ with a supervision plateform can be useful. 

Do you known if you can configuration NagStaMon to talk to the shinken webui ? The list of supported servers does not mention shinken. Here is the code used for various backends : https://github.com/HenriWahl/Nagstamon/tree/master/Nagstamon/Nagstamon/Server

What would be a good solution for this ? 

* the solution is already in the webui, use URL XXXX
* add code to webui to answer an existing API call 
* add code to nagstamon to add a shinken backend
* add a Thruk (or other web UI that uses livestatus) in front of shinken and point to that

**Comment by mohierf** [comment]
It may be a nice idea ... I will have a look at NagStaMon

Currently you may use the Dashboard one eye view in the WebUI ... this view has been designed to be left opened on a fullscreen browser to keep an eye on your system and get alerted about the main problems. Did you try it ?

**Comment by arthurlogilab** [comment]
@mohierf yep, I did see this view. I'm looking for something more discreet that turns red. But thanks for the suggestion.

**Comment by mohierf** [comment]
I will contact the people from NagStaMon to discuss about the best integration strategy ... and I will keep you informed in this issue.

**Comment by Solvik** [comment]
:+1:

**Comment by SylvainGuibert** [comment]
@mohierf : Good idea. tell us about your exchange with nagstamon. thanks :+1:

**Comment by mohierf** [comment]
I had some discussion with the people at NagStaMon and it may be possible to integrate the two applications, no problem for this ... it is only a matter of time! 

 By the way, a huge refactoring is on process for NagStaMon and I think it may be interesting to wait a little for his new version ...

 Stand by for the moment ...

**Comment by SylvainGuibert** [comment]
@mohierf : Thanks for your feedback. :+1:

**Comment by taziden** [comment]
Hi, do you have any news from Nagstamon refactoring and shinken-webui support ?

Thanks

**Comment by SylvainGuibert** [comment]
Nagstamon 2.0 still in BETA [2.0 BETA-20160429] Version but they release a new version each week. Follow it on twitter : https://twitter.com/Nagstamon

**Comment by mohierf** [comment]
Yes. Still in beta version ... and this feature is not a very often required one.

Please :+1:  on this issue if you are interested with this feature for I can evaluate the interest.

**Comment by taziden** [comment]
:+1:

**Comment by SylvainGuibert** [comment]
:+1:

**Comment by ddurieux** [comment]
Seems a good idea for people not have a dedicated monitor with webui status page ;)

**Comment by achaussier** [comment]
:+1:

**Comment by guerini** [comment]
:+1:

**Comment by peacand** [comment]
:+1:

**Comment by JeanRev** [comment]
:+1:

**Comment by maethor** [comment]
Problem is, instead of proposing a common format (in JSON, XML, whatever), Nagstamon is parsing full web pages. We will not re-create a false Thruk or Icinga page in the WebUI just for Nagstamon to parse.

I believe if someone wants to add Shinken WebUI support in Nagstamon, just ask them to propose a monitor for a simple, standardized API, and we will be happy to add a route that returns the correct JSON or XML.

**Comment by HenriWahl** [comment]
Nagstamon started with website parsing but meanwhile every supported monitor delivers its data in a more useful format like JSON/XML/whatever. So you can easily export your data into Nagstamon without any Web UI.

**Comment by maethor** [comment]
@HenriWahl do you have a link to a documentation about this?

**Comment by HenriWahl** [comment]
https://github.com/HenriWahl/Nagstamon/tree/master/Nagstamon/Servers - if you look closer only Nagios web pages still are parsed, the others provide useful data.

**Comment by maethor** [comment]
Source code is not what I call a documentation. Sorry but I don't have time and motivation to read and understand this.

**Comment by mohierf** [comment]
I agree with @maethor ... I already had a look at this source code and it is not that easy to understand what may be implemented on the server side to build a NagStaMon server plugin :confused: 

@HenriWahl : do you have some few doc that would help to define what must be done for login/logout and getting states ? With such a doc it will be quite easy to set up a kind of REST interface in the Shinken WebUI ...

**Comment by HenriWahl** [comment]
@maethor @mohierf I am sorry, this is all available documentation.

As every monitoring solution treats things pretty differently than others they are all very flexible - and different. The GUI part of Nagstamon expects from a server object (derived from GenericServer, which is basically a Nagios server as template) some dictionaries full of information about host/service states - how it retrieved them does not matter.

Seen the other way round - how would Shinken like to hand out its data? I am sure it might be possible to write a Shinken module, because there is even a module for Livestatus (https://github.com/HenriWahl/Nagstamon/blob/master/Nagstamon/Servers/Livestatus.py) which as far as I know does not fit into the otherwise typical webserver based scheme.

**Comment by mohierf** [comment]
@HenriWahl: Ok.

You are right:
- the Shinken livestatus module does not fit at all for such an interface
- it may be possible to write a module for Shinken, but we would adapt the current WebUI rather than rewriting a module

The best solution is to modify the WebUI to create some kind of REST API that would be used by one of your Server plugin. But, currently, it is mainly a matter of time ...

**Comment by HenriWahl** [comment]
Yes, if there is already a WebUI with authentication it seems the most useful way to extend it to provide its data in your favorite format.

**Comment by xsylx** [comment]
Nagstamon has released 2.01 stable version.

Some news about Webui2 compatibilty with Nagstatmon ?

**Comment by jackewitz** [comment]
Magstamon 3.0.2 is out. What about connect shinken to nagstamon?

**Comment by maethor** [comment]
Hello everyone,

I need help on this. If you want, I can make an URL that returns JSON. This is easy for me. But I need somebody to point me to Nagstamon documentation or just to tell me what format is expected, and to do what is necessary on Nagstamon side (tests, PR, I don't know). I don't have time to dig in Nagstamon source code or to contact the Nagstamon team since I am not using Nagstamon myself.

Thanks,
